p2p/conn/connection.go

```diff
@@ -200,8 +203,6 @@ func (c *MConnection) OnStart() error {
 	if err := c.BaseService.OnStart(); err != nil {
 		return err
 	}
-	c.quitSendRoutine = make(chan struct{})
```

Comment:
why move? I thought initializing vars in OnStart() is safer since a service can be restarted, although we don't do it

Reply:
Oh you're right. I wanted to avoid checking it for nil but can do something else there instead

Reply:
Ok fixed. Those should never actually be nil by the time we're calling a stop function so we should be ok.